scripts/generate_enums.py

```diff
@@ -0,0 +1,356 @@
+#! /usr/bin/env python3
+import argparse
+import os
+import string
+import textwrap
+import xml.etree.ElementTree as ElementTree
+
+CONFIG_FILE_PATH = "../build/docs/public_api/xml/config_8h.xml"
+
+ROC_JAVA_BASE_PATH = "../../roc-java"
+ROC_GO_BASE_PATH = "../../roc-go"
+
+ROC_JAVA_PACKAGE = "org.rocstreaming.roctoolkit"
+
+ODD_PREFIXES = {'roc_clock_source': 'ROC_CLOCK_', 'roc_protocol': 'ROC_PROTO_'}
+
+
+def to_camel_case(name):
+    return ''.join(x.capitalize() for x in name.split('_'))
+
+
+class DocComment:
+    def __init__(self, items: list):
+        self.items: list = items
+
+
+class EnumValue:
+    def __init__(self, name: string, value: string, doc: DocComment):
+        self.name: string = name
+        self.value: string = value
+        self.doc: DocComment = doc
+
+
+class EnumDefinition:
+    def __init__(self, name: string, values: list[EnumValue], doc: DocComment):
+        self.name: string = name
+        self.values: list[EnumValue] = values
+        self.doc: DocComment = doc
+
+
+class EnumGenerator:
+    def generate_enum(self, enum_definition: EnumDefinition):
+        pass
```

Comment:
nit: convention is to raise NotImplementedError

Reply:
done